telemetry/definitions/commonDefinitions.json

```diff
@@ -4322,6 +4330,48 @@
             ],
             "passive": true
         },
+        {
+            "name": "codewhisperer_codeScanChatFileScan",
+            "description": "Called when file scan is selected from chat",
+            "metadata": [
+                {
+                    "type": "codeScanSessionId",
+                    "required": false
+                },
+                {
+                    "type": "credentialSourceId",
+                    "required": false
+                }
+            ]
+        },
+        {
+            "name": "codewhisperer_codeScanChatNewTab",
+            "description": "Called when a new chat tab is opened in the code scan view",
+            "metadata": [
+                {
+                    "type": "codeScanSessionId",
+                    "required": false
+                },
+                {
+                    "type": "credentialSourceId",
+                    "required": false
+                }
+            ]
+        },
+        {
+            "name": "codewhisperer_codeScanChatProjectScan",
+            "description": "Called when project scan is selected from chat",
```

Comment:
this metric and `codewhisperer_codeScanIssueIgnore` look over-specific. it's not maintainable or scalable to create a metric for every tiny variation. instead, consider a generalized metric like

```
codewhisperer_codeScanScan {
  source = 'chat' | ...
  type = 'file' | 'project'
}
```

If that's not acceptable, at least avoid baking `Chat` in the metric name. It's not unlikely that other "sources" will trigger a scan in the future, and it is again costly to define a metric for each and every scenario, which a more generic metric could instead be parameterized. That also helps dashboard maintenance...

Reply:
Makes sense, updating